os/fs/smartfs/smartfs_smart.c

```diff
@@ -693,7 +753,66 @@ static int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_of
 	}
 #endif							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+	/* File's data sector has been synced with MTD, now check if this is a new file entry and write the entry */
+	if (sf->bflags && SMARTFS_BFLAG_NEW_ENTRY) {
+		/* Allocate memory for buffer space to write entry to MTD later and return if space can't be allocated */
+		entrysize = sizeof(struct smartfs_entry_header_s) + fs->fs_llformat.namesize;
+		tmp_buf = (char *)kmm_malloc(entrysize);
+		if (tmp_buf == NULL) {
+			fdbg("Unable to allocate memory\n");
+			ret = -ENOMEM;
+			goto errout;
+		}
+		memset(tmp_buf, CONFIG_SMARTFS_ERASEDSTATE, entrysize);
+		entry = (struct smartfs_entry_header_s *)&tmp_buf[0];
+
+		/* Populate the directory entry to be written in the parent's sector */
+
+#ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
+		smartfs_wrle16(&entry->firstsector, sf->entry.firstsector);
+#ifdef CONFIG_SMARTFS_ENTRY_TIMESTAMP
+		smartfs_wrle32((uint8_t*)&entry->utc, fs->entry_seq++);
+#else
+		smartfs_wrle32((uint8_t*)&entry->utc, time(NULL));
+#endif
+#else
+		entry->firstsector = sf->entry.firstsector;
+#ifdef CONFIG_SMARTFS_ENTRY_TIMESTAMP
+		entry->utc = fs->entry_seq++;
```

Comment:
so, open -> sync will inrease sequence number 2 times.
can't we do this kind of 'set attribute value of entry' in find_available_entry ?

Reply:
Modified